spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractEndpoint.java

```diff
@@ -63,6 +69,23 @@ public void setTaskScheduler(TaskScheduler taskScheduler) {
 		super.setTaskScheduler(taskScheduler);
 	}
 
+	@Override
+	protected void onInit() throws Exception {
+		super.onInit();
+
+		if (!this.autoStartupSetExplicitly) {
+			String[] endpointNamePatterns =
+					getIntegrationProperty(IntegrationProperties.ENDPOINTS_NO_AUTO_STARTUP, String[].class);
+
+			for (String pattern : endpointNamePatterns) {
+				if (PatternMatchUtils.simpleMatch(pattern, getComponentName())) {
+					this.autoStartup = false;
+					break;
+				}
+			}
```

Comment:
That's good idea, @olegz , but it works only for the `master`, where we already have Java 8.
Since we have to backport the feature to the `4.3.x`, still based on Java 7, we don't have choice unless continue to live with old iteration style here.

OTOH I'm always a bit suspicious for Java Stream API usage  since it can't leave without objects creating and garbage collection, meanwhile the simple iteration with the condition statements are just logical levers without any allocations.